In [11]:
import seaborn as sns
import pandas as pd
import numpy as np
# Load dataset
tips = sns.load_dataset('tips')

# Show first 5 rows
tips.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


### Here we see in the info that:
* The dataset contains 244 rows and 7 columns.
* Variables include both numerical (total_bill, tip, size) and categorical (sex, smoker, day, time) features.
* There are no missing values.

In [5]:
tips.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

### Columns details in tips : 
- total_bill -> Total bill (USD) 
- tip -> Tip amount (USD) 
- sex -> Gender of the bill payer 
- smoker -> Whether the person smokes 
- day -> Day of the week 
- time -> Lunch or Dinner 
- size -> Number of people at the table 

In [7]:
print(tips.describe())


       total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000


### Interpretation:
- 
- Mean tip is around $3.00, with most tips between $2–$4.
- Party sizes range from 1 to 6, with an average size of 2.57

In [8]:
print(tips.isnull().sum())


total_bill    0
tip           0
sex           0
smoker        0
day           0
time          0
size          0
dtype: int64
